packages/quantic/force-app/main/default/lwc/quanticSort/quanticSort.js

```diff
@@ -148,6 +232,28 @@ export default class QuanticSort extends LightningElement {
   }
 
   get value() {
-    return this.state?.sortCriteria;
+    return this.hasCustomSortOptions
+      ? this.customSortOptions[0].value
+      : this.state?.sortCriteria;
+  }
```

Comment:
why this change here? 

Theoretically the value should always come from the state.

Reply:
because if we pass slots, we want the first option passed to be selected otherwise it defaults at relevancy even in a case where relevancy is not one of the passed slots.

ie:
<img width="290" alt="image" src="https://github.com/coveo/ui-kit/assets/73316533/405e82fa-89f3-4bd0-8b12-d5cb693f0d99">

Reply:
but are you also reflecting this in the Headless state?

Reply:
double checked, it is reflected in state, no need for that so I reverted!

Reply:
how is the state reflected?
I don't see any logic the gives the Headless library the information about the custom sorts available. 

To achieve such logic, we need to use the `initialState` key passed in the `props` property of the buildSort controller: https://github.com/coveo/ui-kit/blob/a1aba3e05763a123e7fdadf99318601887df9a0d/packages/headless/src/controllers/core/sort/headless-core-sort.ts#L23-L28.

Currently, that's behaviour we have:

https://github.com/coveo/ui-kit/assets/86681870/b6df791c-4d09-4af3-be7b-79282c9da887

which is wrong, the state needs to be always in sync with what we have to display to avoid such bugs.

Please add the necessary logic to fix this bug and add the e2e tests to cover that.

Reply:
this was addressed by adding a `initialState` prop to the headless build engine.

I also added E2E tests to cover this edge case and added logic to make sure the sort options are shared properly between main page ones and refine modal ones.